test/parallel/test-dns.js

```diff
@@ -62,6 +62,32 @@ assert(existing.length > 0);
   ]);
 }
 
+{
+  // Various invalidities, all of which should throw a clean error.
+  const invalidServers = [
+    ' ',
+    '\n',
+    '\0',
+    '1'.repeat(3 * 4),
+    // Check for REDOS issues.
+    ':'.repeat(100000),
+    '['.repeat(100000),
+    '['.repeat(100000) + ']'.repeat(100000) + 'a'
+  ];
+  invalidServers.forEach((serv) => {
+    assert.throws(
+      () => {
+        dns.setServers([serv]);
+      },
+      {
+        name: 'TypeError [ERR_INVALID_IP_ADDRESS]',
+        code: 'ERR_INVALID_IP_ADDRESS'
+      },
+      `Unexpected error thrown for serv '${serv}'`
```

Comment:
Nit: It would seem that the most likely case for this assertion to fire is that `dns.setServers()` does not throw here, rather than it throws the wrong error type. In that case, the output of the `AssertionError` will be:

```console
AssertionError [ERR_ASSERTION]: Missing expected exception: Unexpected error thrown for serv  
```

Especially given that a few of `serv` values are whitespace, I think maybe just omitting the third argument to `assert.throws()` altogether might be better. If it were just this:

```js
assert.throws(
    () => {
      dns.setServers([serv]);
    },
    {
      name: 'TypeError [ERR_INVALID_IP_ADDRESS]',
      code: 'ERR_INVALID_IP_ADDRESS'
    }
);
```

...then the default message would display and that one is pretty clear I think:

```console
AssertionError [ERR_ASSERTION]: Missing expected exception.
```

It also improves the output when the wrong type of error is thrown:

```console
AssertionError [ERR_ASSERTION]: Input A expected to strictly deep-equal input B:
+ expected - actual

  Comparison {
-   name: 'RangeError',
-   code: 'ERR_OOPS'
+   name: 'TypeError [ERR_INVALID_IP_ADDRESS]',
+   code: 'ERR_INVALID_IP_ADDRESS'
  }
```

With the code currently in this branch/PR, the user would get this instead:

```console
AssertionError [ERR_ASSERTION]: Unexpected error thrown for serv  
```